cupy/random/sample.py

```diff
@@ -136,4 +139,52 @@ def random_sample(size=None, dtype=float):
     return rs.random_sample(size=size, dtype=dtype)
 
 
-# TODO(okuta): Implement choice
+def choice(a, size=None, replace=True, p=None):
+    """Return an array of random values from a given 1-D array.
+
+    Each element of returned values are independently sampled
+    from a according to p or uniformly.
+
+    Args:
+        a (1-D array-like or int):
+            If an ndarray, a random sample is generated from its elements.
+            If an int, the random sample is generated as if a was
+            cupy.arange(n)
+        size (int or tuple of ints): The shape of the array.
+        replace (boolean): Whether the sample is with or without replacement
+        p (1-D array-like):
+            The probabilities associated with each entry in a.
+            If not given the sample assumes a uniform distribution over all
+            entries in a.
+
+    Returns:
+        cupy.ndarray: An array of a values distributed according to p or
+                      uniformly.
+
+    .. seealso:: :func:`numpy.random.choice`
+
+    """
+    a = cupy.array(a, copy=False)
+    if a.ndim == 0:
+        try:
+            pop_size = operator.index(a.item())
+        except TypeError:
+            raise ValueError("a must be 1-dimensional or an integer")
+        if pop_size <= 0:
+            raise ValueError("a must be greater than 0")
+    elif a.ndim != 1:
+        raise ValueError("a must be 1-dimensional")
+    else:
+        pop_size = a.shape[0]
+        if pop_size is 0:
```

Comment:
Is there any reason why we use `is` for comparing integers? If not, `pop_size == 0` is better.